File: Year-2023/LeetCode-July-Challenge/Pow.java
class Solution {
    public double myPow(double x, long n) {
        double res = 1;
        if(n == 0 || x == 1) return 1;
        else if(n < 0){
            x = 1 / x;
            n = -n;
        }

        res *= x;
        n--;
        int t = 0;

        while(n != 0){
            t++;
            if(t < n){
                res *= res;
                n -= t;
                t += t;
            } if(n > 0){
                res *= x;
                n--;
            }
        }
        return res;
    }
}
public class Pow{
    public static void main(String[] args) {
        double x = 2.00000; long n = 10;
        System.out.println(new Solution().myPow(x, n));
    }
}

File: Year-2023/LeetCode-July-Challenge/Pow.js
var myPow = function (x, n) {
    let res = 1;
    if (n === 0 || x === 1) return 1;
    else if (n < 0) {
        x = 1 / x;
        n = -n;
    }

    res *= x;
    n--;
    let t = 0;

    while (n) {
        t++;
        if (t < n) {
            res *= res;
            n -= t;
            t += t;
        } if (n > 0) {
            res *= x;
            n--;
        }
    }
    return res;
};

let x = 2.00000, n = 10;
console.log(myPow(x, n));